mage/main.go

```diff
@@ -99,33 +111,45 @@ func ParseAndRun(dir string, stdout, stderr io.Writer, stdin io.Reader, args []s
 		log.Println("Build Date:", timestamp)
 		log.Println("Commit:", commitHash)
 		return 0
-	}
-	if mageInit {
+	case INIT:
 		if err := generateInit(dir); err != nil {
 			log.Println("Error:", err)
 			return 1
 		}
 		log.Println(initFile, "created")
 		return 0
+	case CLEAN:
+		dir := mg.CacheDir()
+		if err := removeContents(dir); err != nil {
+			log.Println("Error:", err)
+			return 1
+		}
+		log.Println(dir, "cleaned")
+		return 0
 	}
```

Comment:
can you add 
```
case None:
    return Invoke(inv)
default:
   panic(fmt.Errorf("Unknown command type: %v", cmd)
}
```

That way we will make sure to catch if we add any new commands.

Also, list should be a command (the -l flag)

Reply:
Would you mind if I kept the updating of `list` to a command as a separate PR?